crates/host/src/secrets.rs

```diff
@@ -26,8 +26,9 @@ pub(crate) struct SecretReference {
     /// The version of the secret to retrieve. If not supplied, the latest version will be used.
     #[serde(default, skip_serializing_if = "Option::is_none")]
     pub version: Option<String>,
-    /// The policy that defines configuration options for the backend.
-    pub policy_properties: String,
+    /// The policy that defines configuration options for the backend. This is a serialized
+    /// JSON object that will be passed to the backend for policy evaluation.
+    pub policy: String,
```

Comment:
```suggestion
    pub policy_json: String,
```

A bit late but...

Reply:
Agreed on making this a `serde_json::Value`... but why isn't this just a `BTreeMap`/`HashMap`? Would it save validation down the line?

Reply:
I didn't change this one because we want it to be `policy` downstream, so I'd have to change the name and then add a serde rename